ts/features/bonus/siciliaVola/screens/voucherList/VoucherDetailsScreen.tsx

```diff
@@ -188,6 +167,29 @@ const VoucherDetailsScreen = (props: Props): React.ReactElement | null => {
     );
   }
 
+  // eslint-disable-next-line react-hooks/rules-of-hooks
+  const { present, bottomSheet, dismiss } = useIOBottomSheet(
+    <VoucherDetailBottomSheet
+      barCode={selectedVoucher.barCode}
+      qrCode={selectedVoucher.qrCode}
+      onExit={() => dismiss()}
+      onSaveVoucher={() => {
+        dismiss();
+        props.stampaVoucher(selectedVoucher.id);
+      }}
+      pdfVoucherState={pdfVoucherState}
+    />,
+    I18n.t("bonus.sv.components.voucherBottomsheet.title"),
+    650
+  );
+
+  const openQrButtonProps = {
+    primary: true,
+    bordered: false,
+    onPress: present,
+    title: I18n.t("bonus.sv.voucherList.details.cta.openQr")
+  };
+
```

Comment:
With the current implementation, the hook is positioned in a conditional branch, violating the rule of hooks https://en.reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level. This generates the following error: 
<img width="398" alt="Schermata 2022-04-13 alle 19 17 29" src="https://user-images.githubusercontent.com/26501317/163235091-8a8a0a9d-21f7-4880-aaf2-e853d516d17c.png">

Here my proposal that should fix the problem, moving the hook before the conditional part:
```typescript
useEffect(() => {
    fold(
      pdfVoucherState,
      () => null,
      () => null,
      _ => showToast(I18n.t("bonus.sv.pdfVoucher.toast.ok"), "success"),
      _ => showToast(I18n.t("bonus.sv.pdfVoucher.toast.ko"))
    );
  }, [pdfVoucherState]);

  const { present, bottomSheet, dismiss } = useIOBottomSheet(
    isReady(props.selectedVoucher) ? (
      <VoucherDetailBottomSheet
        barCode={props.selectedVoucher.value.barCode}
        qrCode={props.selectedVoucher.value.qrCode}
        onExit={() => dismiss()}
        onSaveVoucher={() => {
          dismiss();
          props.stampaVoucher(selectedVoucher.id);
        }}
        pdfVoucherState={pdfVoucherState}
      />
    ) : null,
    I18n.t("bonus.sv.components.voucherBottomsheet.title"),
    650
  );

  // The selectedVoucherCode can't be undefined in this screen
  if (!isReady(props.selectedVoucher)) {
    ... etc...
  }
```